File: src/backend/SchoolBudget/Entities/Fund.cs
namespace SchoolBudget.Entities;

public enum FundType
{
    Default = 0,
    Targeted = 1,
    Continuous = 2,
}

public class Fund : BaseEntity<Fund>
{
    public string Name { get; set; } = null!;
    public FundType Type { get; init; }
    public bool IsClosed { get; private set; }

    public bool CanClose => Type == FundType.Targeted;

    public void Close()
    {
        if (!CanClose)
            throw new InvalidOperationException();

        IsClosed = true;
    }
}

File: src/frontend/school-budget/src/models/fund.ts
import { BaseEntity } from "./base-entity";

export enum FundType
{
    Default = 'Default',
    Targeted = 'Targeted',
    Continuous = 'Continuous',
}

export interface Fund extends BaseEntity {
  name: string;
  type: string;
  isClosed: boolean;

  canClose: boolean;
}
